### PR TITLE
Explicitly set the minimum requirement for glib2

### PR DESCRIPTION
**Pull request by sgallagher**
There is a segfault that occurs on RHEL/CentOS 7.0 systems. We
need to ensure that if someone builds on that platform, they do so
with a fixed glib2.

### CODE REVIEW

File: tools/cockpit.spec [hunk 1]
```diff
@@ -52,6 +52,10 @@ BuildRequires: docbook-style-xsl
 BuildRequires: keyutils-libs-devel
 BuildRequires: dbus-devel
 BuildRequires: glib-networking
+
+# There is a segfault in g_source_add_child_source that was fixed in
+# 2.37.3-2-g1d5c815
+BuildRequires: glib2-devel >= 2.40
```
Comment (stefwalter) [inline tools/cockpit.spec]: We already check the build time requirement in configure.ac, so I don't think this one isn't necessary.

Comment (fabiand) [inline tools/cockpit.spec]: IIUIC this spec change is necessary to automatically get yum to pull in the updated glibc2.

because, even if glibc2 is available as an update, but the user just installs cockpit, currently the updated glibc2 will not be instaled/updated.

Comment (stefwalter) [inline tools/cockpit.spec]: Yes, I was commenting on the BuildRequires line. Is that change necessary? The Requires: line is good.

Comment (sgallagher) [inline tools/cockpit.spec]: Hmm, it looks like configure.ac does do an explicit check for 2.37.4. That would help ensure that it *fails* if the buildroot has an older version, but doesn't force it to be pulled in if your build system happens to have the older version. (And for things like mock, it will fail earlier as the build-dep installation will fail instead of downloading and installing the older packages, then running configure to fail).

We should probably make sure these are the same value though. The upstream change appeared to happen in 2.37.3, but I'm fine with making this patch require 2.37.4 instead to match the existing configure.ac check. It's unlikely to make a difference.

Comment (sgallagher) [inline tools/cockpit.spec]: Also, it appears that I have 2.40 in one place and 2.37.3 in another. I started with 2.40 then narrowed it down, but forgot to change both locations...

**Comment by sgallagher** [comment]
Updated patch sets both the BuildRequires: and Requires: to glib2 2.37.4 to match the configure.ac check.

**Comment by stefwalter** [comment]
Pulled into master. Moved the comment into the commit message.
